nashvillefcc/nashvillefcc2020website
# Events improved error handling#88

### Description of Changes

- Added new logic for error handling with events data.
 1. Checks for the status of the promise which is returned from the `pingEvents` function.
 2. If, after 5 seconds, the status is still `ok` but the events list is empty, then we timeout and set the status of the events to `error`
- Added `react-spinners` library and added a loading spinner, so that users get a visual cue during events loading.
- Re-added the `connect-logo-container` class which centers the connect logo with the main content. This can be removed if it causes issues, however I have yet to find any.
- Changed the way we determine if an event is online or in-person because one of the most recent events was displayed incorrectly.

### How can I test this?

1. Clone the files from github onto your local workspace.
2. Run `npm install` to install needed dependencies.
3. Once finished, test to make sure the Netlify CLI is installed by typing `netlify` in your console. If not type `npm install netlify-cli -g` in the console to install it.
4. Once confirmed, in index.js uncomment  `'http://localhost:8888/.netlify/functions/pingEvents'` to access the Netlify function.
5. Then type in `netlify dev` to start the development server.
6. Then you are ready to test!
7. Go to `localhost:8000` to see the spinner and error timeout occur.
